src/internal/observable/of.ts

```diff
@@ -79,5 +80,5 @@ export function of<A extends readonly unknown[]>(...values: A): Observable<Value
  */
 export function of<T>(...args: Array<T | SchedulerLike>): Observable<T> {
   const scheduler = popScheduler(args);
-  return from(args as T[], scheduler);
+  return scheduler ? scheduled(args as T[], scheduler) : from(args as T[]);
```

Comment:
Maybe `fromArrayLike` would be better here?

```suggestion
  return scheduler ? scheduled(args as T[], scheduler) : fromArrayLike(args as T[]);
```